source/common/network/listener_filter_buffer_impl.h

```diff
@@ -56,14 +60,16 @@ class ListenerFilterBufferImpl : public ListenerFilterBuffer, Logger::Loggable<L
   ListenerFilterBufferOnCloseCb on_close_cb_;
   ListenerFilterBufferOnDataCb on_data_cb_;
 
-  // The buffer for the data peeked from the socket.
-  std::unique_ptr<uint8_t[]> buffer_;
-  // The start of buffer.
-  uint8_t* base_;
+  bool on_data_cb_disabled_{};
   // The size of buffer;
   uint64_t buffer_size_;
   // The size of valid data.
   uint64_t data_size_{0};
+
+  // The buffer for the data peeked from the socket.
+  std::unique_ptr<uint8_t[]> buffer_;
+  // The start of buffer.
+  uint8_t* base_;
```

Comment:
This change the order of some fields, is that necessary?

Reply:
Yes because buffer_size_ needs to be evaluated first in construstor